llvm/lib/CAS/HashMappedTrie.cpp

```diff
@@ -155,30 +156,23 @@ TrieSubtrie *TrieSubtrie::sink(
 }
 
 struct ThreadSafeHashMappedTrieBase::ImplType {
-  static std::unique_ptr<ImplType> create(size_t StartBit, size_t NumBits) {
+  static ImplType *create(size_t StartBit, size_t NumBits) {
     size_t Size = sizeof(ImplType) + getTrieTailSize(StartBit, NumBits);
     void *Memory = ::malloc(Size);
-    ImplType *Impl = ::new (Memory) ImplType(StartBit, NumBits);
-    return std::unique_ptr<ImplType>(Impl);
+    return ::new (Memory) ImplType(StartBit, NumBits);
   }
 
   TrieSubtrie *save(std::unique_ptr<TrieSubtrie> S) {
-    TrieSubtrie *Ptr = S.get();
+    TrieSubtrie *Ptr = S.release();
```

Comment:
Do we need this variable any more? I think you can just use `S.get()` throughout, then release only at the end. The `Ptr` variable was a necessity because of all the awkward ownership transfers but I think it obfuscates what's going on.